### PR TITLE
Add LuaCheck and fix warnings

### PR DESCRIPTION
**Pull request by Panquesito7**
Things added/changed:

- Add LuaCheck and fix warnings.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+on: [push, pull_request]
+name: build
+jobs:
+  lint:
```
Comment (Niklp09) [inline .github/workflows/build.yml:4]: Should be called `luacheck` (see https://github.com/mt-mods/plantlife_modpack/pull/53#discussion_r1270833014)

File: .luacheckrc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+unused_args = false
+max_line_length = false
+
+globals = {
+    "coloredwood"
+}
+
+read_globals = {
+    table = {fields = {"copy", "getn", "contains"}},
```
Comment (Niklp09) [inline .luacheckrc:9]: coloredwood uses only `table.copy` and `table.insert`

File: .luacheckrc [hunk 3]
```diff
@@ -0,0 +1,13 @@
+unused_args = false
+max_line_length = false
```
Comment (Niklp09) [inline .luacheckrc:2]: Not needed

**Comment by Niklp09** [comment]
@Panquesito7 You marked my comments as "resolved" but didn't push any changes.

**Comment by OgelGames** [comment]
FYI, we use this luacheck now: https://github.com/lunarmodules/luacheck (it's faster)

**Comment by Panquesito7** [comment]
> @Panquesito7 You marked my comments as "resolved" but didn't push any changes.

Sorry, that's strange. I thought I pushed changes. 😅

**Comment by Panquesito7** [comment]
> FYI, we use this luacheck now: https://github.com/lunarmodules/luacheck (it's faster)

This workflow, right? https://github.com/lunarmodules/luacheck/blob/master/.github/workflows/luacheck.yml

**Comment by OgelGames** [comment]
> This workflow, right? https://github.com/lunarmodules/luacheck/blob/master/.github/workflows/luacheck.yml

No, it's a slightly modified version of the example in the readme. (using the master branch instead of a fixed version)

**Pull request closed without merging**